distsql/request_builder.go

```diff
@@ -245,6 +261,49 @@ func (builder *RequestBuilder) SetTiDBServerID(serverID uint64) *RequestBuilder
 	return builder
 }
 
+// SetFromInfoSchema sets the following fields from infoSchema:
+// "bundles"
+func (builder *RequestBuilder) SetFromInfoSchema(is infoschema.InfoSchema) *RequestBuilder {
+	if is == nil {
+		return builder
+	}
+	builder.bundles = is.RuleBundles()
+	return builder
+}
+
+func (builder *RequestBuilder) verifyTxnScope() error {
+	if builder.txnScope == "" {
+		builder.txnScope = oracle.GlobalTxnScope
+	}
+	if builder.txnScope == oracle.GlobalTxnScope || len(builder.bundles) < 1 {
+		return nil
+	}
+	visitTableID := make(map[int64]struct{})
+	for _, keyRange := range builder.Request.KeyRanges {
+		tableID := tablecodec.DecodeTableID(keyRange.StartKey)
+		if tableID > 0 {
+			visitTableID[tableID] = struct{}{}
+		} else {
+			return fmt.Errorf("requestBuilder can't decode tableID from keyRange")
```

Comment:
We'd better keep error stack message? Maybe `errors.New()` would be better

Reply:
updated.